provider/aws/aws.go

```diff
@@ -849,27 +849,32 @@ func useAlias(ep *endpoint.Endpoint, preferCNAME bool) bool {
 	return false
 }
 
-// isAWSAlias determines if a given endpoint is supposed to create an AWS Alias record
-// and (if so) returns the target hosted zone ID
-func isAWSAlias(ep *endpoint.Endpoint) string {
-	prop, exists := ep.GetProviderSpecificProperty(providerSpecificAlias)
-	if exists && prop.Value == "true" && ep.RecordType == endpoint.RecordTypeCNAME && len(ep.Targets) > 0 {
-		// alias records can only point to canonical hosted zones (e.g. to ELBs) or other records in the same zone
-
-		if hostedZoneID, ok := ep.GetProviderSpecificProperty(providerSpecificTargetHostedZone); ok {
-			// existing Endpoint where we got the target hosted zone from the Route53 data
-			return hostedZoneID.Value
-		}
+// aliasHostedZone returns the target hosted zone ID for an ALIAS record or the empty string if not an ALIAS
+func aliasHostedZone(ep *endpoint.Endpoint) string {
+	if !isAlias(ep) {
+		return ""
+	}
+	// alias records can only point to canonical hosted zones (e.g. to ELBs) or other records in the same zone
 
-		// check if the target is in a canonical hosted zone
-		if canonicalHostedZone := canonicalHostedZone(ep.Targets[0]); canonicalHostedZone != "" {
-			return canonicalHostedZone
-		}
+	if hostedZoneID, ok := ep.GetProviderSpecificProperty(providerSpecificTargetHostedZone); ok {
+		// existing Endpoint where we got the target hosted zone from the Route53 data
+		return hostedZoneID.Value
+	}
 
-		// if not, target needs to be in the same zone
-		return sameZoneAlias
+	// check if the target is in a canonical hosted zone
+	if canonicalHostedZone := canonicalHostedZone(ep.Targets[0]); canonicalHostedZone != "" {
+		return canonicalHostedZone
 	}
-	return ""
+
+	// if not, target needs to be in the same zone
+	return sameZoneAlias
+}
+
+// isAlias tests whether an endpoint is an ALIS. Note this does not do ALIAS inference here (see inferAlias), which is
+// unnecessary after AWSProvider.AdjustEndpoints.
+func isAlias(ep *endpoint.Endpoint) bool {
+	prop, _ := ep.GetProviderSpecificProperty(providerSpecificAlias)
```

Comment:
we do not check if the value exists, I'm not sure if is this intended?

Reply:
It makes no difference to the behavior and it makes the logic simpler.

If it doesn't exist, prop.Value is the empty string and the function will return false, as it would if we explicitly checked whether it exists.

This is the same as the higher level business rule, which is that alias is only true if set precisely true -- otherwise it's false. There's no other state for when it's present but set to some other value.